builtin/mainmenu/tab_online.lua

```diff
@@ -264,19 +265,49 @@ end
 
 --------------------------------------------------------------------------------
 
+local function parse_search_input(input)
+	-- Return nil if nothing to search for
+	if not input:find("%S") then
+		return nil
+	end
+
+	local query = {keywords = {}, mods = {}, players = {}}
+
+	-- Process quotation enclosed parts
```

Comment:
> Tokenize into words using` :gmatch("%S+")` as below

This does not work, e.g. `"Your   Land"` would become `"Your`, `Land"`. (spaces are lost)

Before your suggestion, I used  `:gmatch("\"([^\"]*)\"([^\"]*)")`[^1] to separate every quoted and non-quoted part.
I guess this is essentially what you are suggestion now.

However, I think the current solution is good enough since it is easier to read, even though it may be a little slower because the string gets reassembled. 

[^1]: missing check for space before and after quotation mark

Reply:
> (spaces are lost)

Yeah, bit of a brainfart on my part, sorry. The current solution is okay. I suppose we could reassemble with the spaces but eh, that's a mess too.

For the future: Maybe we should start doing proper recursive descent parsers earlier.